### DISCUSSION TITLE
Determining Token Orientation on a PDF page

### DISCUSSION THREAD

**Discussion by rudra0713** [pymupdf/PyMuPDF#3153] [Q&A]
Hi,  I am curious to know whether it's possible to determine a token's orientation (for example: vertical or horizontal) on a given PDF page using PyMuPDF.

**Comment by JorjMcKie** [comment]
A typical "Discussions" post!
Transferring ...

**Comment by JorjMcKie** [comment]
What do you mean by token? A piece of text? You can find the writing direction ...
If you mean vector graphics: you can see whether the area of a rectangle is rotated left or right ...

**Comment by rudra0713** [comment]
Sorry, I meant every span of text in a block (as per the definition of block dictionaries, mentioned here https://pymupdf.readthedocs.io/en/latest/textpage.html#structure-of-dictionary-outputs).

**Reply by JorjMcKie** [reply]
Ok, thanks.
The span [dictionary](https://pymupdf.readthedocs.io/en/latest/textpage.html#span-dictionary), together with the [line](https://pymupdf.readthedocs.io/en/latest/textpage.html#line-dictionary) dictionary gives you the writing direction and (almost) everything else you may want to know about the text span ...

**Reply by JorjMcKie** [reply]
Do you have questions about details there?

**Reply by JorjMcKie** [reply]
The only actually **_missing_** information is whether a span contains a right-to-left text. I am working on including this also.
